### PR TITLE
[EHN] Inclusion of SincShallowNet

### PR DESCRIPTION
**Pull request by bruAristimunha**

### CODE REVIEW

File: braindecode/models/sinc_shallow.py [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+
+    - The sinc-convolutional layer initializes cutoff frequencies uniformly
+        within the desired frequency range and updates them during training while
+        ensuring the lower cutoff is less than the upper cutoff.
+
+    Parameters
+    ----------
+    num_temp_filters : int
+        Number of temporal filters in the SincFilter layer.
+    temp_filter_size : int
```
Comment (sylvchev) [inline braindecode/models/sinc_shallow.py:52]:
```suggestion
    time_filter_len : int
```
You could use `time_filter_len` or  `temporal_filter_size` as `temp_filter_size` seems like is temporary. Could also be `time_kernel_size`.

File: braindecode/models/sinc_shallow.py [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+
+    **Implementation Notes:**
+
+    - The sinc-convolutional layer initializes cutoff frequencies uniformly
+        within the desired frequency range and updates them during training while
+        ensuring the lower cutoff is less than the upper cutoff.
+
+    Parameters
+    ----------
+    num_temp_filters : int
```
Comment (sylvchev) [inline braindecode/models/sinc_shallow.py:50]:
```suggestion
    num_time_filters : int
```
`temp_filters` is misleading, might be understood as temporary. Better use `time` or `temporal`.

File: braindecode/models/sinc_shallow.py [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+        low = self.n_pi * low_freqs  # [kernel_size // 2, num_filters]
+        high = self.n_pi * high_freqs  # [kernel_size // 2, num_filters]
+
+        filters_left = (torch.sin(high) - torch.sin(low)) / (self.n_pi / 2.0)
+        filters_left *= self.window
+        filters_left /= 2.0 * bandwidths
+
+        filters_left = filters_left.unsqueeze(0).unsqueeze(
+            2
+        )  # [1, kernel_size // 2, 1, num_filters]
```
Comment (sylvchev) [inline braindecode/models/sinc_shallow.py:303]:
```suggestion
        # [1, kernel_size // 2, 1, num_filters]
        filters_left = filters_left.unsqueeze(0).unsqueeze(2)  
```

File: braindecode/models/sinc_shallow.py [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+        ----------
+        inputs : torch.Tensor
+            Input tensor of shape [batch_size, num_channels, num_samples, 1].
+
+        Returns
+        -------
+        torch.Tensor
+            Filtered output tensor of shape [batch_size, num_channels, num_samples, num_filters].
+        """
+        # torch.Size([1, 32, 256, 1]) vs TensorShape([1, 32, 256, 1])
```
Comment (sylvchev) [inline braindecode/models/sinc_shallow.py:326]:
```suggestion
```

File: docs/whats_new.rst [hunk 5]
```diff
@@ -57,6 +57,7 @@ Enhancements
 - Adding :class:`braindecode.models.CTNet`  (:gh:`666` by `Bruno Aristimunha`_ )
 - Fix warnings not being suppressed when creating a TUHAbnormal dataset in parallel (:gh:`670` by `itsaphel`_)
 - Exposing :class:`braindecode.models.EEGITNet` hyper-parameters (:gh:`672` by `Bruno Aristimunha`_)
+- Adding :class:`braindecode.models.SincShallowNet` (:gh:`XXX` by `Bruno Aristimunha`_ )
```
Comment (sylvchev) [inline docs/whats_new.rst:60]:
```suggestion
- Adding :class:`braindecode.models.SincShallowNet` (:gh:`678` by `Bruno Aristimunha`_ )
```

**Review by sylvchev — APPROVED** [review]

File: braindecode/models/sinc_shallow.py [hunk 6]
```diff
@@ -0,0 +1,338 @@
...
+        """Builds the sinc filters based on current parameters."""
+        # Computing the low frequencies of the filters
+        low_freqs = self.min_freq + torch.abs(self.low_freqs)
+        # Setting a minimum band and minimum freq
+        high_freqs = torch.clamp(
+            low_freqs + torch.abs(self.bandwidths),
+            min=self.min_freq,
+            max=self.sfreq / 2.0,
+        )
+        bandwidths = high_freqs - low_freqs
```
Comment (bruAristimunha) [inline braindecode/models/sinc_shallow.py:291]: normalization matters here. 

Get the model after the training and before (check if the random stage initialization is the same).

Comment (bruAristimunha) [reply braindecode/models/sinc_shallow.py:291]: Already solve on EEGMiner.
https://github.com/braindecode/braindecode/blob/cf53ebc1e1805913c5331dfc0711d78f2e300f12/braindecode/models/eegminer.py#L388
